lib/network_utils.pm

```diff
@@ -44,7 +44,10 @@ sub setup_static_network {
     assert_script_run qq(echo -e "\\nSTARTMODE='auto'\\nBOOTPROTO='static'\\nIPADDR='$ip'\\nMTU='$mtu'">/etc/sysconfig/network/ifcfg-$iface);
     assert_script_run 'rcnetwork restart';
     assert_script_run 'ip addr';
-    assert_script_run "ping -c 1 $gw || journalctl -b --no-pager -o short-precise > /dev/$serialdev";
+    if (script_run("ping -c 1 $gw") != 0) {
+        script_run("journalctl -b --no-pager -o short-precise > /dev/$serialdev");
+        die("Gateway is not reachable, please check your network configuration and setups");
+    }
```

Comment:
How about just a clean

```suggestion
    assert_script_run 'ping -c 1 $gw', fail_message => 'Gateway is not reachable, please check your network configuration and setups';
```

and leave the journalctl call to the post_fail_hook which should/might already handle the case of getting the system journal in case of no network?

Reply:
Thanks, it makes sense!